internal/services/v1alpha1/v1alpharevision.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	"encoding/base64"
+	"fmt"
+	"sort"
+
+	"github.com/shopspring/decimal"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/authzed/spicedb/internal/datastore"
+	internal "github.com/authzed/spicedb/internal/proto/impl/v1"
+)
+
+const (
+	errEncodeError = "error encoding revision: %w"
+	errDecodeError = "error decoding revision: %w"
+)
+
+// computeV1Alpha1Revision computes and encodes a string representing the revisions for each
+// of the namespaces in the map.
+func computeV1Alpha1Revision(revisions map[string]datastore.Revision) (string, error) {
+	msg := &internal.V1Alpha1Revision{}
+	for nsName, revision := range revisions {
+		msg.NsRevisions = append(msg.NsRevisions, &internal.NamespaceAndRevision{
+			NamespaceName: nsName,
+			Revision:      revision.String(),
+		})
+	}
+
+	sort.Sort(byName(msg.NsRevisions))
+
+	// NOTE: We use a Deterministic marshaller here to ensure the produced revision is the same
+	// across calls. Also note that this is not *guarenteed* to be the same across versions of
```

Comment:
spelling -> guaranteed

this is also kinda scary although I'm not sure if makes too much of a difference for our use cases

Reply:
Yeah